### PR TITLE
Restructure Code

### LINKED ISSUE #42: Restructure code
`dns.c` is doing too many things -- strategically break out functionality into smaller library files.

### LINKED ISSUE #45: Change header guards in `dns.h`
Change from `REQUEST_H` -> `DNS_H`

### PR DESCRIPTION
**Pull request by zrottman**
Curious for feedback as to how I've clustered functionality thus far before proceeding further. Does this seem like a reasonable way to organize the code?

### CODE REVIEW

File: src/dns.c [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#include <stdlib.h>
+
```
Comment (orgnizedmess) [inline src/dns.c:1]: Re-add `<stdlib.h>` as running the code without it returned errors.

File: src/dns.h [hunk 2]
```diff
@@ -60,40 +66,61 @@ typedef struct DNSPacket
...
+ * ------------------------------------------- */
+static DNSPacket*   NewDNSPacket(const unsigned char *response_bytes);
+static int          destroy_DNSPacket(DNSPacket** packet);
+static void         display_DNSPacket(DNSPacket *packet);
+
+/* -------------------------------------------
+ * DNS Record
+ * ------------------------------------------- */
+static int          destroy_DNSRecord(DNSRecord** record);
+static void         display_DNSRecord(DNSRecord *record);
```
Comment (orgnizedmess) [inline src/dns.h:115]: From a structure point of view, Packet = Header + Questions + Records, so the DNS Packet functions could be placed after DNS Record

**Review by orgnizedmess — CHANGES_REQUESTED** [review]
I've suggested two changes, everything else looks good to me

**Comment by zrottman** [comment]
As discussed, noting here that, to avoid a painful merge with recent changes to `main`, we should abandon this branch and create a new branch from `main` for this issue.

**Comment by zrottman** [comment]
Replaced by #55

**Pull request closed without merging**